From alan@linux.intel.com  Mon Oct 12 15:01:52 2009
From: Alan Cox <alan@linux.intel.com>
Date: Mon, 12 Oct 2009 15:38:26 +0100
Subject: Staging: et131x: Fix the add_10bit macro
To: greg@kroah.com, linux-kernel@vger.kernel.org
Message-ID: <20091012143823.20944.3745.stgit@localhost.localdomain>


Duh.. we need to preserve the wrap bit when adding.

Signed-off-by: Alan Cox <alan@linux.intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/staging/et131x/et1310_address_map.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/staging/et131x/et1310_address_map.h
+++ b/drivers/staging/et131x/et1310_address_map.h
@@ -223,7 +223,7 @@ typedef union _TXDMA_PR_NUM_DES_t {
 
 extern inline void add_10bit(u32 *v, int n)
 {
-	*v = INDEX10(*v + n);
+	*v = INDEX10(*v + n) | (*v & ET_DMA10_WRAP);
 }
 
 /*
